src/CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+set(definitions "UNDERSCORE" "NFILES=32" "MAXCD=250" "MAXNC=600" "MXNAF=3")
+add_compile_definitions(definitions)
+
+if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
+  set (FREAL8 "-r8")
+  set (FINT8 "-i8")
+  set (TRACEBACK "-traceback")
+  set (MMMED "-mcmodel=medium")
+  set (IntelComp TRUE)
+elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set (FREAL8 "-fdefault-double-8" "-fdefault-real-8")
+  set (FINT8 "-fdefault-integer-8")
+  set (TRACEBACK "-fbacktrace")
+  set (MMMED "")
+endif()
+
+
+set(fortran_8_flags ${FREAL8} ${FINT8} ${TRACEBACK})
+set(shared_flags  ${DEBINFO} ${FOPT3} ${AVX2})
```

Comment:
what is DEBINFO, FOPT3, AVX2? 
And why would one use AVX2 with GNU compiler?

Reply:
There are mtune options for the gnu compiler that match up with AVX2

Reply:
regardless, they are not defined.
Why cant we reduce this whole down to something like this:
```
if(CMAKE_C_COMPILER_ID MATCHES "^(Intel)$")
  set(c_DA_allocation_def "DYNAMIC_ALLOCATION")
  set(c_SA_allocation_def "STATIC_ALLOCATION")
  set(c_8_definitions "F77_INTSIZE_8")
  set(c_8_DA_definitions "F77_INTSIZE_8")
  set(CMAKE_Fortran_FLAGS "-g -traceback")
  set(CMAKE_Fortran_FLAGS_DEBUG "-O0")
  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
  set(fortran_8_flags "-r8 -i8")
  set(fortran_d_flags "-r8")
  set(fortran_8_DA_flags "-r8 -i8")
  set(fortran_d_DA_flags "-r8")
elseif(CMAKE_C_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
  set(CMAKE_C_FLAGS_DEBUG "-ggdb -Wall -O0")
  set(CMAKE_Fortran_FLAGS "-g -fbacktrace -funroll-loops")
  set(CMAKE_Fortran_FLAGS_DEBUG "-ggdb -Wall -O0")
  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
  set(fortran_8_flags "-fdefault-real-8 -fdefault-integer-8")
  set(fortran_d_flags "-fdefault-real-8")
endif()
```